components/ThemeSwitch.tsx

```diff
@@ -0,0 +1,26 @@
+"use client";
+
+import { useState, useEffect } from "react";
+import { useTheme } from "next-themes";
+
+export const ThemeSwitch = () => {
+  const [mounted, setMounted] = useState(false);
+  const { theme, setTheme } = useTheme();
+
+  // useEffect only runs on the client, so now we can safely show the UI
+  useEffect(() => {
+    setMounted(true);
+  }, []);
```

Comment:
why do we need this?

Reply:
To show a component on the client side only. This might be useful for Stackly too.